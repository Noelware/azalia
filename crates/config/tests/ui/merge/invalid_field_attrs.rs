// üêª‚Äç‚ùÑÔ∏èü™ö azalia: Noelware's Rust commons library.
// Copyright (c) 2024-2025 Noelware, LLC. <team@noelware.org>
//
// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"), to deal
// in the Software without restriction, including without limitation the rights
// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
// copies of the Software, and to permit persons to whom the Software is
// furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in all
// copies or substantial portions of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
// SOFTWARE.

#[derive(azalia_config::merge::Merge)]
pub struct A {
    #[merge(skip)]
    #[merge(skip)]
    a: bool,
}

#[derive(azalia_config::merge::Merge)]
pub struct B {
    #[merge(strategy = "x + 1")]
    b: u32,
}

#[derive(azalia_config::merge::Merge)]
pub struct C {
    #[merge(strategy = 1234)]
    c: String,
}

#[derive(azalia_config::merge::Merge)]
pub struct D {
    #[merge(strategy = azalia_config::merge::strategy::f32::with_negatives)]
    #[merge(strategy = azalia_config::merge::strategy::f32::with_negatives)]
    d: f32,
}

#[derive(azalia_config::merge::Merge)]
pub struct E {
    #[merge(skip = "data")]
    e: usize,
}

#[derive(azalia_config::merge::Merge)]
pub struct F {
    #[merge(unknown_field)]
    f: f64,
}

fn main() {}
