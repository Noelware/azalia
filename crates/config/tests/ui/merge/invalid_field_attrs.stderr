error: field already has `#[merge(skip)]`
  --> tests/ui/merge/invalid_field_attrs.rs:25:13
   |
25 |     #[merge(skip)]
   |             ^^^^

error: unexpected token
  --> tests/ui/merge/invalid_field_attrs.rs:31:24
   |
31 |     #[merge(strategy = "x + 1")]
   |                        ^^^^^^^

error: expected either a qualified path (i.e, `std::mem::replace`) or a literal string that can be a qualified path
  --> tests/ui/merge/invalid_field_attrs.rs:37:24
   |
37 |     #[merge(strategy = 1234)]
   |                        ^^^^

error: field already has `#[merge(strategy)]`
  --> tests/ui/merge/invalid_field_attrs.rs:44:13
   |
44 |     #[merge(strategy = azalia_config::merge::strategy::f32::with_negatives)]
   |             ^^^^^^^^

error: expected `,`
  --> tests/ui/merge/invalid_field_attrs.rs:50:18
   |
50 |     #[merge(skip = "data")]
   |                  ^

error: unknown field, expected either `skip`, `strategy`
  --> tests/ui/merge/invalid_field_attrs.rs:56:13
   |
56 |     #[merge(unknown_field)]
   |             ^^^^^^^^^^^^^
